### ISSUE TITLE
RangeError: Maximum call stack size exceeded 

### ISSUE THREAD

**Issue by flipps-dev** [meteor/meteor#9586]
Hello everyone, I'm struggling with this error. After the meteor installation on windows 10 I wanted to try if it worked, but when I type in the terminal `meteor --version` I get this error:

`C:\Users\Users\AppData\Local\.meteor\packages\meteor-tool\1.6.1\mt-os.windows.x86_64\tools\fs\files.js:59
var useParsedSourceMap = function (pathForSourceMap) {
                                  ^
RangeError: Maximum call stack size exceeded`

I thought the issue is in the file.js, but I don't know how I can fix it.

**Comment by abernix** [comment]
Please see https://github.com/meteor/meteor/issues/7896#issuecomment-344399399, which I believe is a duplicate of this issue.  Other developers have reported success by moving other (higher up in the directory structure) `node_modules` out of the way.

If you could go through the same steps of moving other `node_modules` directories out of the way, but saving them, I'd certainly be interested in knowing exactly which turned out to be the offender.  Perhaps it's the case that we need to change the `meteor-tool` itself (not for apps) to not use the same default module resolution paths.  Though the fact that this is happening still leads me to believe that something else is slightly awry.  Unfortunately, I haven't been able to put my finger on it yet (or reproduce it myself).

**Comment by flipps-dev** [comment]
A question,does it have to be a meteor folder inside %AppData%\Roaming\npm\node_module?

**Comment by abernix** [comment]
No, however that could be a problematic directory so I'd recommend temporarily renaming it to anything besides `node_modules` temporarily in order to eliminate it as a possibility.  Additionally, any of the directories listed in the output of `meteor node -p "modules.paths"`, when run from your app directory, could be a problem.  The full plan for resolving modules, is [quite complicated](https://nodejs.org/api/modules.html#modules_all_together), but hopefully you'll find something quickly in the places I've suggested.

**Comment by flipps-dev** [comment]
It works yeah! I had another node_modules folder at C:\Users\UserName\node_modules I renamed this one to node_modules_old and created another one named node_modules. Maybe this last step is not important, maybe i had just renamed the old node_modules folder, without ctreate a new one, But now meteor works! Thank you!

**Comment by abernix** [comment]
Can you give me some idea of what was inside that `node_modules` directory (now `node_modules_old`)?  If possible, a ZIP-file of its contents might be helpful in developing a fix for it!  This isn't the first time this has affected a developer, and it's likely worth investigating a better solution.

**Comment by flipps-dev** [comment]
Of course, how can I reach you out?

**Comment by abernix** [comment]
If you think it could contain sensitive information (it shouldn't), you can private message me on [the Meteor forums](https://forums.meteor.com/).  I'm also `abernix` on there.

Alternatively, perhaps a link to the ZIP on a Dropbox-like-service?

**Comment by flipps-dev** [comment]
[https://mega.nz/#!4LJEnQqC!BYYns-8AljRCyNvlRU6Vq2xtLaujY26AzMzXXe-oXDo](url)

**Comment by flipps-dev** [comment]
`https://mega.nz/#!4LJEnQqC!BYYns-8AljRCyNvlRU6Vq2xtLaujY26AzMzXXe-oXDo`

**Comment by flipps-dev** [comment]
[node_modules_old.zip](https://github.com/meteor/meteor/files/1678358/node_modules_old.zip)

**Comment by abernix** [comment]
I can't be sure I'll get to this right away (esp. as it'll likely wait until I'm on Windows again) but thank you for providing that artifact.  It could be very helpful to resolving this issue!

**Comment by juho** [comment]
FWIW we encounter this when deploying on Linux, solved with the environment variable 
```
export NODE_OPTIONS=--max_old_space_size=4096
```
There's some discussion about it on the forums here: https://forums.meteor.com/t/how-to-increase-decrease-node-js-memory-limits-using-meteor/34355

**Comment by hwillson** [comment]
Since this issue was resolved in https://github.com/meteor/meteor/issues/9586#issuecomment-361578921 (and can also be addressed by increasing `max_old_space_size` as mentioned in https://github.com/meteor/meteor/issues/9586#issuecomment-367965066), I'll close it for now. If for some reason this issue becomes more popular, we'll definitely re-open and investigate further. Thanks!

**Comment by yehudahkay** [comment]
Hi I'm having this issue as well, can't really use meteor, I tried installing on windows 10 and get 

C:\node_modules\source-map-support\source-map-support.js:51
  return function(arg) {
                 ^

RangeError: Maximum call stack size exceeded

I have also npm installed previously but changing the name of node_modules didn't help

**Comment by yehudahkay** [comment]
I changed the name of C:\node_modules and it worked, but is very slow compared to a mac i tested it on. I tried installing on windows subsystem for linux but it is even slower. I use other packages on windows like git, npm, exp and ionic and they all work in git bash i don't need to open a windows cmd, and they also never had any conflicts. If meteor could work automatically in that fashion it would be great.

**Comment by durjoy28** [comment]
![screenshot 41](https://user-images.githubusercontent.com/20689241/47011425-c01e9f80-d163-11e8-947a-828d56aa6867.png)

**Comment by yehudahkay** [comment]
@durjoy28 try the solutions we posted, but if you have a mac i found meteor was designed more for mac...

**Comment by durjoy28** [comment]
@yehudahkay I've no mac but right now I need the meteor. I tried by changing node modules name but it not working. Please let me know what should I do?

**Comment by yehudahkay** [comment]
Im sorry I cant help any more, you can actually run a mac virtual machine on windows also. Search for the issue there are a lot of recommended solutions.

**Comment by durjoy28** [comment]
My problem is solved. Just, I changed my another node_modules folder C:\Users\Durjoy\node_modules_old previously it was C:\Users\Durjoy\node_modules. I shared for others as if they can solve easily when they will face the same problem.

**Comment by nja12** [comment]
This solution worked for me too, thanks all

**Comment by preyasha2810** [comment]
@durjoy28 hello! have u solved the error???
![image](https://user-images.githubusercontent.com/45910402/53423884-354e8b00-3a08-11e9-9e55-722381349bd8.png)

**Comment by durjoy28** [comment]
Dear sir, still problem is appear. Available solution is not working for
me.

**Comment by edemaine** [comment]
I just ran into this issue, unaware that I'd accidentally made a `C:\Users\myusername\node_modules` folder, and surprised that that stopped `meteor` from successfully running (in a different directory, `C:\Users\myusername\Projects\myprojectname`).  The fix works, but presumably there's an underlying issue that causes this to happen?  Can we fix it, instead of everyone needing to find this thread and delete/rename that folder?

**Comment by zodern** [comment]
There is an open issue for the cause: https://github.com/meteor/meteor/issues/10930#issuecomment-678580436. It should be fixed once #11740 is merged and the Windows build machine is updated to publish the meteor-tool with working symlinks.

**Comment by LizhenQiao** [comment]
> I just ran into this issue, unaware that I'd accidentally made a `C:\Users\myusername\node_modules` folder, and surprised that that stopped `meteor` from successfully running (in a different directory, `C:\Users\myusername\Projects\myprojectname`). The fix works, but presumably there's an underlying issue that causes this to happen? Can we fix it, instead of everyone needing to find this thread and delete/rename that folder?

Vote for the idea. I was new to a meteor project and run into this issue. Though it is simple to change the folder name. It still took me several hours to find this solution. And to be honest, I believe it will prevent a great number of users who have the interest to get involved in meteor project. BTW, even simply change the name of `node_modules` folder itself looks uncomfortable to me. (
